### ISSUE TITLE
Modules not found when bower is not used as package manager

### ISSUE THREAD

**Issue by nicoabie** [nwolverson/purescript-language-server#58]
First of all thanks for the language server.

I'm having an issue going through the tutorials of https://github.com/adkelley/javascript-to-purescript. 
The dependencies gets installed in the .psc-package folder instead of the bower_components
I believe the problem is that the server assumes (and It was correct in the past) that the deps are in the bower_components folder, see
https://github.com/nwolverson/purescript-language-server/blob/5390f83300f6f1f3ec8eeb6110d14fff809e1bf6/src/LanguageServer/IdePurescript/Config.purs#L60

If that is the issue maybe I can provide some help with your guidance, I'm trying to learn PS.

**Comment by nwolverson** [comment]
You don't post any detailed error, and no idea what editor or tool you are using with the language server, so hard to help, but "modules not found" sounds like nothing will be working for you - completion, tooltips, etc. The package path there is then irrelevant, what matters is just that you have compiled the project prior to launching the language server (ie your IDE) and paths etc are set up correctly (`purs` at the right version in your path), and the language server is launched with the right source folders (open your project folder with the `bower.json` or `psc-package.json` at the root level).

The setting you see for package paths  is only required for a couple of things though - getting documentation when showing completions etc, and navigating to source. If `psc-package` is used, a setting is available `addPscPackageSources` which will add the correct psc-package source directories, similarly `addSpagoSources` for `spago` - if you're using vscode, for example, this will appear with some similar setting in the config UI.

**Comment by nicoabie** [comment]
I've provided a [link](https://github.com/adkelley/javascript-to-purescript) to the repository to be able to replicate the issue but it is true that I did not specify the IDE I'm using. It's vscode. Sorry for that.

The projects have both a bower.json and a psc-package.json (I don't know if this is a good idea, coming from js feels it's like having a package-lock and a yarn-lock)

When installing the dependencies using `psc-package install` all deps gets intalled in the .psc-package folfer.

If I also run `bower install` (which makes no sense because I have everything already installed by psc) the language server works as expected.

I noticed in the Purescript configuration that there is a Package Path pointing to bower_components.

If I point it to `.psc-package/psc-0.13.3` I can remove the bower_components folder and everything works.

Is that correct? When using psc instead of bower the Package Path should be updated? Thanks in advance.

**Comment by nwolverson** [comment]
You're right that you don't want to use both `psc-package` and `bower`, so you can just ignore the `bower.json`, nothing wrong should happen due to it existing. If you install both with bower and psc-package, I could believe odd things happen.

The setting you want in vscode is "Add Psc Package Sources":
![image](https://user-images.githubusercontent.com/2770891/66405533-c7c0ff00-e9d9-11e9-9cfb-e1eb5557350c.png)
The "Package Path" setting also can work as you describe, but this is really only required for more custom scenarios, the "Add Psc Package Sources" will not require updating if you change package set version, for example.

**Comment by OuyangQianba** [comment]
Hi, @nwolverson 
I'm using coc.nvim and spago. I have this issue too.
I tried use `--config` to set `addSpagoSources` to true like this :

![image](https://user-images.githubusercontent.com/17270099/69388392-ac991d00-0d03-11ea-9bab-0039e942f37b.png)

I also tried to set `packagePath ` to `.spage`, `buildCommand` to `spago build`

But this problem always there.
![image](https://user-images.githubusercontent.com/17270099/69388381-a145f180-0d03-11ea-97af-6c68f2645b45.png)

Could you help on this please?
